packages/mobile/src/transactions/feed/TransactionFeed.tsx

```diff
@@ -131,11 +93,106 @@ function mapStandbyTransactionToFeedTokenTransaction(tx: StandbyTransaction): Fe
   }
 }
 
+function addFetchedTransactionFunction(
+  fetchedTransactions: TokenTransaction[],
+  setFetchedTransactions: any
+) {
+  return (transactions: TokenTransaction[]) => {
+    const currentHashes = new Set(fetchedTransactions.map((tx) => tx.transactionHash))
+
+    const transactionsWithoutDuplicatedHash = fetchedTransactions.concat(
+      transactions.filter((tx) => !currentHashes.has(tx.transactionHash))
+    )
+
+    transactionsWithoutDuplicatedHash.sort((a, b) => {
+      return b.timestamp - a.timestamp
+    })
+
+    setFetchedTransactions(transactionsWithoutDuplicatedHash)
+  }
+}
+
 function TransactionFeed() {
+  const address = useSelector(walletAddressSelector)
+  const localCurrencyCode = useSelector(getLocalCurrencyCode)
+
+  const [pageInfo, setPageInfo] = useState<PageInfo | null>(null)
+  const [fetchMoreTransactions, setFetchMoreTransactions] = useState(false)
+
   const cachedTransactions = useSelector(transactionsSelector)
+  const [fetchedTransactions, setFetchedTransactions] = useState<TokenTransaction[]>([])
+  const addFetchedTransactions = addFetchedTransactionFunction(
+    fetchedTransactions,
+    setFetchedTransactions
+  )
+
+  const dispatch = useDispatch()
+
+  // Update the counter variable every |POLL_INTERVAL| so that a query is made to the backend.
+  const [counter, setCounter] = useState(0)
+  useInterval(() => setCounter((n) => n + 1), POLL_INTERVAL)
+
+  const { loading, error } = useAsyncQueryTransactionsFeed({
```

Comment:
i'm confused about why this hook exists twice in this component, is this intentional? is it possible to have it called only once?

Reply:
I have the same confusion, aren't we making duplicate requests?